### PR TITLE
[TEVA-1002] more than one school job location

### PR DESCRIPTION
**Pull request by josephhull676**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-1002

## Changes in this PR:

- Migrated job_location to use integers
- Created job_location enum and job_location_options helper
- Removed redundant test and validation
- Split options helpers in VacanciesHelpers module into a new VacanciesOptionsHelpers module. Included VacanciesOptionsHelpers in the VacanciesHelpers module.

Radio Buttons and error message:

Screenshot 2020-08-19 at 12 27 43

### CODE REVIEW

**Comment by david-mears-dfe** [comment]
> You don't need to do the migration because the database doesn't have any data yet.

I can see how this applies to the conversion of legacy data, but we do need to migrate job_location from string to integer if we're to use enums.

**Comment by david-mears-dfe** [comment]
> I'm unsure how to change the order of the radio buttons. I can see they're different in the figma. 'At more than one school in the trust' is second in the figma.

This is determined by the order of JOB_LOCATION_OPTIONS (or whatever method is transforming that) - the radio buttons are made based on an array passed into govuk_collection_radio_buttons

File: app/helpers/vacancies_options_helper.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+module VacanciesOptionsHelper
+  def job_location_options
```
Comment (david-mears-dfe) [inline app/helpers/vacancies_options_helper.rb:2]: Mild preference for these to be in alphabetical order for easy finding, if all else is equal

Comment (josephhull676) [reply app/helpers/vacancies_options_helper.rb:2]: Will do that now sir!

**Comment by alexbowen** [comment]
seems rather light on tests, well a bit lighter than before as one has been removed?!

**Comment by david-mears-dfe** [comment]
Regardless of whether the flag is added by this PR or a separate one, you will need to use it to prevent the new option being displayed.

File: app/helpers/vacancies_options_helper.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Vacancy::JOB_LOCATION_OPTIONS.map do |key, _value|
+      [I18n.t("helpers.fieldset.job_location_form.job_location_options.#{key}"), key]
+    end
+  end
+
+  def job_role_options
+    options = Vacancy::JOB_ROLE_OPTIONS.except(:nqt_suitable).map do |key, _value|
+      [I18n.t("jobs.job_role_options.#{key}"), key]
+    end
+    options.reject { |option| option == 'at_multiple_schools' } unless MultiSchoolJobsFeature.enabled?
```
Comment (cpjmcquillan) [inline app/helpers/vacancies_options_helper.rb:12]: There should be a spec for this helper method asserting whether or not the option is included depending on if the feature is enabled or not

Comment (josephhull676) [reply app/helpers/vacancies_options_helper.rb:12]: Will add the test now

File: app/helpers/vacancies_options_helper.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+module VacanciesOptionsHelper
+  def hired_status_options
+    Vacancy.hired_statuses.keys.map { |k| [t("jobs.feedback.hired_status.#{k}"), k] }
+  end
+
+  def job_location_options
```
Comment (cpjmcquillan) [inline app/helpers/vacancies_options_helper.rb:6]:
```ruby
Vacancy::JOB_LOCATION_OPTIONS.map { |key, _value| [I18n.t("jobs.job_location_options.#{key}"), key] if MultiSchoolJobsFeature.enabled? || key != :at_multiple_schools  }.reject(&:blank?)
```

File: db/schema.rb [hunk 4]
```diff
@@ -10,14 +10,12 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_08_13_105100) do
+ActiveRecord::Schema.define(version: 2020_08_18_111239) do
 
   # These are extensions that must be enabled in order to support this database
-  enable_extension "citext"
```
Comment (cpjmcquillan) [inline db/schema.rb:16]: You should only commit changes to the schema introduced by the migration itself

File: spec/helpers/vacancies_options_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe VacanciesOptionsHelper, type: :helper do
+  describe '#job_location_options' do
+    context 'when FEATURE_MULTI_SCHOOL_JOBS flag is set to true' do
```
Comment (cpjmcquillan) [inline spec/helpers/vacancies_options_helper_spec.rb:5]:
```ruby
context 'when MultiSchoolJobsFeature is enabled' do
```

File: spec/helpers/vacancies_options_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      it 'returns an array including the multi-school option' do
+        allow(MultiSchoolJobsFeature).to receive(:enabled?).and_return(true)
```
Comment (cpjmcquillan) [inline spec/helpers/vacancies_options_helper_spec.rb:7]: Normally we would do a before block in like this...
```ruby
describe '#job_location_options' do
  before do
    allow(MultiSchoolJobsFeature).to receive(enabled?).and_return(multi_school_jobs_enabled)
  end

  context 'when MultiSchoolJobsFeature is enabled' do
    let(:multi_school_jobs_enabled) { true }

    it 'returns an array including at_multiple_schools' do
      ...
    end
  end

  ...
end
```

File: db/schema.rb [hunk 7]
```diff
@@ -10,15 +10,15 @@
...
+ActiveRecord::Schema.define(version: 2020_08_18_111239) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "citext"
   enable_extension "fuzzystrmatch"
   enable_extension "pgcrypto"
   enable_extension "plpgsql"
   enable_extension "uuid-ossp"
-
+
```
Comment (cpjmcquillan) [inline db/schema.rb:21]: whitespace

File: db/migrate/20200818111239_change_job_location_to_integer_on_vacancies.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+class ChangeJobLocationToIntegerOnVacancies < ActiveRecord::Migration[5.2]
+  def up
```
Comment (cpjmcquillan) [inline db/migrate/20200818111239_change_job_location_to_integer_on_vacancies.rb:2]: What do you think of this instead of just a change method @cesidio ?

**Review by cpjmcquillan — APPROVED** [review]

**Pull request merged**